### ISSUE TITLE
Implement `cache:` field for docker-executor

### ISSUE THREAD

**Issue by bcouetil** [firecow/gitlab-ci-local#106]
# Love 

Thank you very much for this great tool, that will help me a lot with a few more features 🥰

# Request

## Description

Gitlab caching mecanism is ignored. A simple implementation would suffice : always mounting, in docker volumes, the files listed in cache section, for example.

## Symptom

Slow builds

# Workaround

## Description

In the case of Maven, I use :

```
  cache:
    paths:
      - .m2/repository
  variables:
    MAVEN_OPTS: "-Dmaven.repo.local=$CI_PROJECT_DIR/.m2/repository"
```

Populate the .m2 with some commands before lauching the tool :

```
MAVEN_OPTS="-Dmaven.repo.local=.m2/repository" mvn dependency:go-offline --settings ./common/maven.settings.xml
MAVEN_OPTS="-Dmaven.repo.local=.m2/repository" mvn some-particular:runtime-goal --settings ./common/maven.settings.xml
```

And add .m2 to .gitignore

## Impact

- ✅ low : needs actions once before/after a debugging session
- medium : fixed using another top level .gitlab-ci.yml
- high : needs deep refactoring, everywhere in templates or everytime or both

**Comment by firecow** [comment]
I've renamed the card, hope thats ok... But yeah, we need this feature, especially in docker-executor jobs (`image:`)

**Comment by bcouetil** [comment]
Implementing fully-featured cache can be complex. Gitlab zips/unzips and download/upload file, depending on the policy, and this can take significant time at runtime.

As a first and easy step, I tried (yes... I cloned the repo 🎉) to bind mount a cache folder stored under `.gitlab-ci-local/cache` by adding this line in code :
```
dockerCmd += `-v \${PWD}/.gitlab-ci-local/cache/.m2:/builds/.m2 `
```
for this example of cache usage (my main cache usage) :
```
.maven:
  cache:
    key: maven
    paths:
      - .m2
```

This works like a charm, and is super efficient : not a single file copy or move is involved. This can save a lot of time in the pipeline compared to Gitlab way of handling cache !

I suggest this as a first and super efficient implementation. Later, we can implement different (slower) policies, but we could keep this as the fastest implementation with an option like `--ignore-cache-policy` or `--fast-cache`.

What do you think @firecow ?

**Comment by firecow** [comment]
At first glance it sounds like a good idea. `key: files: string[]` cannot work exactly like this, but we could restrict it to "key: string".

**Comment by bcouetil** [comment]
Implemented in #121
